core/iwasm/aot/aot_runtime.c

```diff
@@ -1152,29 +1152,63 @@ init_func_ptrs(AOTModuleInstance *module_inst, AOTModule *module,
                char *error_buf, uint32 error_buf_size)
 {
     uint32 i;
+    AOTImportFunc *import_func;
     void **func_ptrs;
     uint64 total_size = ((uint64)module->import_func_count + module->func_count)
                         * sizeof(void *);
+#if WASM_ENABLE_MULTI_MODULE != 0
+    char *ptr;
+    AOTModuleInstanceExtra *extra;
+    bh_list *sub_module_list_node = NULL;
+    const char *sub_inst_name = NULL;
+#endif
 
     if (module->import_func_count + module->func_count == 0)
         return true;
 
-    /* Allocate memory */
+/* Allocate memory */
+#if WASM_ENABLE_MULTI_MODULE != 0
+    if (!(ptr = runtime_malloc(total_size * 2, error_buf, error_buf_size))) {
```

Comment:
Allocating memory for func_ptrs and func_module_insts together makes code a little confusing, suggest to allocate individual memory for func_module_insts and don't forget to free it when destroying aot instance.

And no need to allocate memory with `import_func_count + func_count`, allocating instances for import functions should be enough.